core/src/test/scala/unit/kafka/server/KafkaApisTest.scala

```diff
@@ -4390,6 +4401,2215 @@ class KafkaApisTest extends Logging {
     assertEquals("broker2", node.host)
   }
 
+  private def expectedAcquiredRecords(firstOffset : Long, lastOffset : Long, deliveryCount : Int) : util.List[AcquiredRecords] = {
+    val acquiredRecordsList : util.List[AcquiredRecords] = new util.ArrayList()
+    acquiredRecordsList.add(new AcquiredRecords()
+      .setFirstOffset(firstOffset)
+      .setLastOffset(lastOffset)
+      .setDeliveryCount(deliveryCount.toShort))
+    acquiredRecordsList
+  }
+
+  private def memoryRecordsBuilder(numOfRecords : Int, startOffset : Long) : MemoryRecordsBuilder = {
+
+    val buffer: ByteBuffer = ByteBuffer.allocate(1024)
+    val compression: Compression = Compression.of(CompressionType.NONE).build()
+    val timestampType: TimestampType = TimestampType.CREATE_TIME
+
+    val builder: MemoryRecordsBuilder = MemoryRecords.builder(buffer, compression, timestampType, startOffset)
+    for (i <- 0 until numOfRecords) {
+      builder.appendWithOffset(startOffset + i, 0L, TestUtils.randomBytes(10), TestUtils.randomBytes(10))
+    }
+    builder
+  }
```

Comment:
Seems the methods are same as defined in `SharePartitionTest`, shall we move them to common test utils (in java)?

Reply:
Yep, the true place for these methods should be in TestUtils file, but both SharePartitionTest and KafkaApisTest import separate TestUtils files. SharePartitionTest use the Java one and KafkaApisTest use the scala one. As far as I know, java does not support aliasing and neither do the older versions of scala.

Reply:
So why not to have either in java or scala test utils which can be used by both?